acme/acme/standalone_test.py

```diff
@@ -88,8 +88,8 @@ def setUp(self):
         }
         from acme.standalone import DVSNIServer
         self.server = DVSNIServer(("", 0), certs=self.certs)
-        # pylint: disable=no-member
-        self.thread = threading.Thread(target=self.server.handle_request)
+        # use serve_forever2() instead of handle_request() to avoid hanging
```

Comment:
Because the `serve_forever2()` has a timeout for `handle_request()`, and this will make sure the function `handle_request()` always returns to its caller. The caller `serve_forever2()` will then check the `self._stopped` flag, and decide whether to call `handle_request()` again.

For the test case, it has to set the timeout to make sure it does not hang if it uses `handle_request()`. In the case `handle_request()` returned, it will need to check whether the test case has done or `handle_request()` timed-out, in the case it timed-out, the test case will need to call the `handle_request()` again. This behaviour exactly matches what `serve_forever2()` function does.